### ISSUE TITLE
Question: Use heed from python

### ISSUE THREAD

**Issue by dineshbvadhia** [meilisearch/heed#236]
Can heed be used from python? If so, are there any examples? Thanks

**Comment by Kerollmops** [comment]
Hey @dineshbvadhia 👋

It is unfortunately not possible to directly find heed on the Python package store. The main reason is that we haven't exposed the C bindings for it.

**Comment by dineshbvadhia** [comment]
Are there plans to expose the C bindings?

**Comment by irevoire** [comment]
What features should it provide that the original lmdb python-wrapper doesn’t provide?

**Comment by dineshbvadhia** [comment]
@irevoire Sorry for the late response. There is a trend to replace critical components in python applications with rust (either as extensions or packages). So, it is not about which features a python-heed would provide over pylmdb.

**Comment by Kerollmops** [comment]
Hey @dineshbvadhia 👋

I think it is not particularly interesting to use heed in Python as the main goal of replacing C libraries with Rust is related to memory safety. Here, heed is a wrapper around a C, non-memory-safe library. Python is already memory-safe, and I think you should use those already existing wrappers.

Furthermore, exposing heed to other libraries requires creating a C interface. It would not make any sense to do that as it will likely result in an interface being similar to LMDB itself as lifetimes, and such securities can't be expressed in Python, resulting in the same library security flows.

In conclusion, I recommend either using the already existing Python LMDB wrappers, switching your whole project to Rust so that you can express lifetimes (and gain performances over Python), or finding a key-value store with a C interface in Rust (but you still lose the lifetime safeties in Python).
